lib/utils/interval/interval.go

```diff
@@ -149,6 +160,16 @@ func (i *Interval) run(timer *time.Timer) {
 			timer.Reset(i.duration())
 			// ensure we don't send any pending ticks
 			ch = nil
+		case <-i.fire:
+			// stop and drain timer
+			if !timer.Stop() {
```

Comment:
I always get confused here, but are you certain this isn't going to deadlock if we've already received from the timer?

https://github.com/golang/go/issues/27169

Reply:
Yes, because we always reset the timer immediately if we read from it.  The linked deadlock is being caused by someone using this pattern on a timer that has fired and been drained.